histogram/histogram_test.go

```diff
@@ -0,0 +1,109 @@
+package histogram
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func Example() {
+	// Create a new descriptor.
+	desc := prometheus.NewDesc(
+		"foo_bar_metric",
+		"The number of foos that have been bared",
+		nil,
+		nil,
+	)
+
+	// Create a Histogram.
+	c := Config{
+		BucketLimits: []float64{0, 1, 5},
+	}
+	h, _ := New(c)
+
+	// Add entries to the Histogram as part of the exporter,
+	// e.g: latency of network connections.
+	h.Add(0.2)
+
+	// Convert the Histogram into a metric.
+	metric := prometheus.MustNewConstHistogram(
```

Comment:
One idea I had was something like:

```
metric := h.Metric(desc) 
```
e.g: a function that takes the descriptor, and returns the metric, to reduce the structs variables visibility, but I'm not sure how necessary it is, and it adds a bit of complexity. Thoughts?

Reply:
Lets start small and iterate.

Reply:
Yeah, didn't need it in the end. getters is enough

Reply:
`metric := h.Metric(desc) ` makes sense to me.